eng/pipelines/templates/steps/sync-directory.yml

```diff
@@ -21,7 +21,7 @@ steps:
   - template: /eng/common/pipelines/templates/steps/create-pull-request.yml
     parameters:
       RepoName: ${{ repo }}
-      PRBranchName: sync-${{ parameters.DirectoryToSync }}
+      PRBranchName: sync-${{ parameters.DirectoryToSync }}-$(Build.BuildNumber)
```

Comment:
We should add the branch name as another template parameter instead of hard-coding this here.